### DISCUSSION TITLE
New Load robot system KukaKR70-R2100 - Problem:  Not Load Robot Model

### DISCUSSION THREAD

**Discussion by AlbertoAlvares** [visose/Robots#211] [Q&A] [answered]
Hi,
I created two files, 3D model (.3dm with Kuka-KR70-R2100 robot model) and DH configuration (.xml), both with the same name, following the script.

I load the Library using "Load robot system" command.

New Robo Kuka is not loaded. To test file configuration error. xml, I use another robot model file that already exists in the library, in this case I tested it with VUB-KR15-c changing the name of the Vub.3dm file to my AlvaresTech.3dm file system and the VUB robot is loaded with my configuration , AlvaresTech.xml .

Why is my 3D model of the new Kuka KR70-R2100 robot not loaded?

Download new Robot Model : https://drive.google.com/file/d/1ep2BTzOHsHWcZHiGEhksRyTiN1VpHlAi/view?usp=sharing

Where am I going wrong?

![AlvaresTech-Kuka-KR70-R2100](https://user-images.githubusercontent.com/1254911/214839660-f58bcf25-48f9-406e-bc18-926fb9062a6e.png)

**Comment by romtecmax** [comment]
i'm having the same issue.

also: is there documentation on how to setup the xml file?
i would like to know what each parameter here means:

`<Revolute number="1" a="56" d="283" minrange="-160" maxrange="160" maxspeed="200" />`

number, max and min range and speed are clear, but how do i set a and d correctly?

**Comment by AlbertoAlvares** [comment, marked as answer]
This paper describes the DH parameters of the Kuka KR family of robots, that is, you only need to edit the xml file of the Robots pro robot Kuka KR70 library
[RoboKukaKR70R2100-ParametrosParaDH (1).pdf](https://github.com/visose/Robots/files/10518587/RoboKukaKR70R2100-ParametrosParaDH.1.pdf)
[Graphical_representation_of_the_significant_6R_KUKA_robots_spaces (1).pdf]
(https://github.com/visose/Robots/files/10518588/Graphical_representation_of_the_significant_6R_KUKA_robots_spaces.1.pdf)

The model of the KukaKR70 R2100 is already functional. Problem was associated with the complexity of the Mesh, Texture, file size with 105 MB. We reduced the complexity, simplifying the mesh, and small file with 1.4 MBytes. Sorted out. You can use the library, as it is functional and could also be included in the online libraries, it could help
[AlvaresTech-Working-KukaKR70-R2100.zip](https://github.com/visose/Robots/files/10518645/AlvaresTech-Working-KukaKR70-R2100.zip)
 other people in the community.

**Reply by visose** [reply]
Hi Alberto, glad that you sorted it out.

Just to clarify, the issue is not the complexity, but that it only supports objects of the type "mesh". It would not have worked either if the geometry was simple but in the form of blocks and polysurfaces.
